### PR TITLE
patch moto to include region header for get and head request and add …

### PR DESCRIPTION
**Pull request by rezafuru**
Add missing bucket region header to appropriate requests 
#4205

### CODE REVIEW

File: localstack/services/s3/s3_starter.py [hunk 1]
```diff
@@ -134,6 +134,34 @@ def get_bucket(self, bucket_name, *args, **kwargs):
...
+        result = _bucket_response_get_orig(self, bucket_name, querystring, *args, **kwargs)
+        # for some reason in the "get-bucket-location" call, moto doesn't return a code, headers, body triple as a result
+        if isinstance(result, tuple) and len(result) == 3:
+            code, headers, body = result
+            bucket = s3_models.s3_backend.get_bucket(bucket_name)
+            headers["x-amz-bucket-region"] = bucket.region_name
+        return result
+
+    _bucket_response_get_orig = s3_responses.ResponseObject._bucket_response_get
+    # method type is only required when the method is part of an object instance (make a method out of a function)
```
Comment (thrau) [inline localstack/services/s3/s3_starter.py:162]: can we also change lines 148-150 then?

**Review by thrau — APPROVED** [review]
LGTM, maybe we could add an issue to move this to moto at some point

File: localstack/services/s3/s3_starter.py (hunk 1 above)
Comment (rezafuru) [reply localstack/services/s3/s3_starter.py:162]: Ah, you mean that we just monkey patch the method and not re-implement it? Sure

File: tests/integration/test_s3.py [hunk 2]
```diff
@@ -2727,3 +2730,23 @@ def _get_test_client(self):
...
+@pytest.fixture
+def s3_client():
+    client = aws_stack.connect_to_service("s3")
+    yield client
+
+
+class TestS3New:
+    def test_region_header_exists(self, s3_client):
+        s3_client.create_bucket(
+            Bucket=TEST_BUCKET_NAME_2,
```
Comment (thrau) [inline tests/integration/test_s3.py:2744]: i think it's better to generate a random bucket name for each test run.

**Comment by thrau** [comment]
excellent!

**Pull request merged**